packages/cw0/src/parse_reply.rs

```diff
@@ -0,0 +1,214 @@
+use thiserror::Error;
+
+use cosmwasm_std::{Binary, Reply};
+
+// Protobuf wire types (https://developers.google.com/protocol-buffers/docs/encoding)
+const WIRE_TYPE_LENGTH_DELIMITED: u8 = 2;
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MsgInstantiateContractResponse {
+    pub contract_address: String,
+    pub data: Option<Binary>,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MsgExecuteContractResponse {
+    pub data: Option<Binary>,
+}
+
+/// Helper function to parse length-prefixed protobuf fields.
+/// The remaining of the data is kept in the data parameter.
+fn parse_protobuf_length_prefixed(data: &mut Vec<u8>) -> Result<Vec<u8>, ParseReplyError> {
+    if data.is_empty() {
+        return Ok(vec![]);
+    };
+    let mut rest_1 = data.split_off(1);
+    let wire_type = data[0] & 0b11;
+    let field = data[0] >> 3;
```

Comment:
When parsing, we know which field we expect.
It would be nice to pass this in as an argument and return an error if it is a different field.

Thus, 

```
let address = parse_protobuf_string(data, 1)?;
let data = parse_protobuf_bytes(data, 2)?;
```

Nice that this returns `Ok(vec![])` if the data was empty (no field 2 present)